Orckestra.Web.Typescript/Orckestra.Web.Typescript/TypescriptHttpModule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Composite.Core.WebClient;
+using Orckestra.Web.Typescript.Classes;
+using System;
+using System.Web;
+
+namespace Orckestra.Web.Typescript
+{
+    public class TypescriptHttpModule : IHttpModule
+    {
+        public static bool IsDebugMode = HttpContext.Current.IsDebuggingEnabled;
+
+        public void Dispose() { }
+        public void Init(HttpApplication application)
+        {
+            if (!Helper.PackageEnabled)
+            {
+                return;
+            }
+            application.BeginRequest += (a, b) => BeginReguest(application.Context);
+        }
+
+        private void BeginReguest(HttpContext httpContext)
+        {
+            string relativeUrl = httpContext.Request.Path;
+            //skipping admin console requests
+            if (string.Equals(relativeUrl, UrlUtils.AdminRootPath, StringComparison.OrdinalIgnoreCase) ||
+                relativeUrl.StartsWith(UrlUtils.AdminRootPath + "/", StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+            TasksPool.CheckSourcesChanges();
```

Comment:
Do we need check for every request? Can we limit it only for js file, or made a handler and execute only for specific target js?

Reply:
Thanks, this is a great idea to decrease influence on a page response speed.
I could provide some check for requests path like:
```
if (httpContext.Request.Path.EndsWith(".js", StringComparison.OrdinalIgnoreCase))
```
In my local tests 1.000.000 executions of `TasksPool.CheckSourcesChanges() `
(source changes check, without compilation in case of changes) take ~60 ms.
1.000.000 executions of *.js check take ~ 15 ms.
So we could save some processor time. Even if `CheckSourcesChanges() `works very fast, if we can save processor time I think this is a great idea.

But the problem is, that a possible result of `CheckSourcesChanges()` is an exception during compilation, which we expected to show (to throw) to a developer if he or she uses a debug mode in web.config. If we throw this exception in *.js request, but not in a main request, it will not shown in a main page window with IIS additional info. We can see it if to use browser debug mode without IIS additional info :( 

As an option, this package can be turned off in settings, not to influence at all, for example when all changes of typescripts are made and js file (or files) are ready.